pkg/remote/aws/launch_template_enumerator.go

```diff
@@ -0,0 +1,65 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/snyk/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/snyk/driftctl/pkg/remote/error"
+	"github.com/snyk/driftctl/pkg/resource"
+	"github.com/snyk/driftctl/pkg/resource/aws"
+)
+
+type LaunchTemplateEnumerator struct {
+	repository repository.EC2Repository
+	factory    resource.ResourceFactory
+}
+
+func NewLaunchTemplateEnumerator(repo repository.EC2Repository, factory resource.ResourceFactory) *LaunchTemplateEnumerator {
+	return &LaunchTemplateEnumerator{
+		repository: repo,
+		factory:    factory,
+	}
+}
+
+func (e *LaunchTemplateEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsLaunchTemplateResourceType
+}
+
+func (e *LaunchTemplateEnumerator) Enumerate() ([]*resource.Resource, error) {
+	templates, err := e.repository.DescribeLaunchTemplates()
+	if err != nil {
+		return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+	}
+
+	results := make([]*resource.Resource, 0, len(templates))
+
+	for _, tmpl := range templates {
+		launchTemplateVersions, err := e.repository.DescribeLaunchTemplateVersions(tmpl)
+		if err != nil {
+			return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+		}
+		ltData := launchTemplateVersions[0].LaunchTemplateData
+
+		results = append(
+			results,
+			e.factory.CreateAbstractResource(
+				string(e.SupportedType()),
+				*tmpl.LaunchTemplateId,
+				map[string]interface{}{
+					"credit_specification": getCreditSpecification(ltData.CreditSpecification),
```

Comment:
I'm not sure to understand why this is required here ?

Reply:
I tested the resource again without it and indeed it is unused, I removed it

Reply:
Yeah, this is not used by any middleware in non deep mode and not in `ReadResource` either this should be OK to drop it.